src/apps/cardano/sign_tx.py

```diff
@@ -118,39 +110,29 @@
         ctx,
         transaction.output_addresses,
         transaction.outgoing_coins,
-        transaction.change_derivation_paths,
-        transaction.change_coins,
         transaction.fee,
-        len(tx_body),
         transaction.network_name,
+        transaction.inputs,
+        transaction.outputs,
     ):
         raise wire.ActionCancelled("Signing cancelled")
 
     return tx
 
 
-def _micro_ada_to_ada(amount: float) -> float:
-    return amount / 1000000
-
-
 class Transaction:
     def __init__(
-        self, inputs: list, outputs: list, transactions: list, root_node, network: int
+        self, inputs: list, outputs: list, transactions: list, root_node, protocol_magic: int
     ):
         self.inputs = inputs
         self.outputs = outputs
         self.transactions = transactions
         self.root_node = root_node
         # attributes have to be always empty in current Cardano
         self.attributes = {}
-        if network == 1:
-            self.network_name = "Testnet"
-            self.network_magic = b"\x01\x1a\x41\x70\xcb\x17\x58\x20"
-        elif network == 2:
-            self.network_name = "Mainnet"
-            self.network_magic = b"\x01\x1a\x2d\x96\x4a\x09\x58\x20"
-        else:
-            raise wire.ProcessError("Unknown network index %d" % network)
+
+        self.network_name = KNOWN_PROTOCOL_MAGICS.get(protocol_magic) or 'Unknown'
```

Comment:
Unless micrypython differs from python here, we should be able specify default value in `get` like this:

```
>>> {'xx': 1}.get('xx', 'def')
1
>>> {'xx': 1}.get('yy', 'def')
'def'
```

Reply:
This works in uPy.

Reply:
fixed